main.go

```diff
@@ -207,6 +207,32 @@ func convertField(curPkg *ProtoPackage, desc *descriptor.FieldDescriptorProto, m
 		return nil, fmt.Errorf("unrecognized field label: %s", desc.GetLabel().String())
 	}
 
+        fieldOptions := desc.GetOptions()
+        if fieldOptions != nil {
+          // is there is a 'required' annotation value on this field?
+          if proto.HasExtension(fieldOptions, protos.E_Require) {
+            requireValue, err := proto.GetExtension(fieldOptions, protos.E_Require)
+            if err == nil {
+              doRequire := *requireValue.(*bool)
+              if (doRequire) {
+                // annotations are telling us to force-require this field, so do it
+                field.Mode = "REQUIRED"
+              }
+            }
+          }
+
+          // is there a type override value on this field?
+          if proto.HasExtension(fieldOptions, protos.E_TypeOverride) {
+            typeValue, err := proto.GetExtension(fieldOptions, protos.E_TypeOverride)
+            if err == nil {
+              typeOverride := *typeValue.(*string)
+              if len(typeOverride) > 0 {
+                field.Type = typeOverride
```

Comment:
simple go implementation of reading the field descriptors/extensions, and applying them